app/questionnaire/router.py

```diff
@@ -117,21 +117,23 @@ def get_next_location_url(
         routing_path: RoutingPath,
         return_to: Optional[str] = None,
         return_to_answer_id: Optional[str] = None,
+        return_to_block_id: Optional[str] = None,
     ) -> str:
         """
         Get the next location in the section. If the section is complete, determine where to go next,
         whether it be a summary, the hub or the next incomplete location.
         """
+        if return_to_url := self._get_return_to_location_url(
+            location,
+            return_to,
+            return_to_answer_id=return_to_answer_id,
+            return_to_block_id=return_to_block_id,
+        ):
+            return return_to_url
+
         if self._progress_store.is_section_complete(
             location.section_id, location.list_item_id
         ):
```

Comment:
So in the suggestion, we were still passing `is_section_complete` as this section will need to know if section is complete anyway, therefore it avoided calling `is_section_complete` twice. i.e here if you have return to section summary, it would evaluate section completeness in get_return_to method and then if section is not completed, it would then revaluate completeness here.

Reply:
Addeed `is_section_complete` again